src/ts/subtitleutils.ts

```diff
@@ -0,0 +1,65 @@
+import {ListSelector, ListSelectorConfig} from './components/listselector';
+import {SubtitleListBox} from './components/subtitlelistbox';
+import {SubtitleSelectBox} from './components/subtitleselectbox';
+
+/**
+ * Helper class to handle all subtitle related events
+ *
+ * This class listens to player events as well as the `ListSelector` event if selection changed
+ */
+export class SubtitleSwitchHandler {
+  private player: bitmovin.PlayerAPI;
+  private listElement: ListSelector<ListSelectorConfig>;
+
+  constructor(player: bitmovin.PlayerAPI, element: SubtitleListBox | SubtitleSelectBox) {
+    this.player = player;
+    this.listElement = element;
+
+    this.bindSelectionEvent();
+    this.bindPlayerEvents();
+    const callback = this.buildUpdateSubtitlesCallback();
```

Comment:
Why do you need a `build` method instead of calling it directly? (Applies to all usages of this method)

Reply:
because when passing the method direct to the player event callbacks we are loosing the lexical scope.

Reply:
You mean the `this` context (binding)?

Reply:
yes

Reply:
The `build*` methods need to be removed. Write the methods directly as arrow functions to get rid of the function return value.